### PR TITLE
[SPARK-7039][SQL]JDBCRDD: Add support on type NVARCHAR

### PR DESCRIPTION
**Pull request by szheng79**
Issue:

https://issues.apache.org/jira/browse/SPARK-7039

Add support to column type NVARCHAR in Sql Server

java.sql.Types:
http://docs.oracle.com/javase/7/docs/api/java/sql/Types.html

### CODE REVIEW

**Comment by srowen** [comment]
@szheng79 Please review https://cwiki.apache.org/confluence/display/SPARK/Contributing+to+Spark -- you need to update the title of this PR.

File: sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRDD.scala [hunk 1]
```diff
@@ -60,6 +60,7 @@ private[sql] object JDBCRDD extends Logging {
       case java.sql.Types.NCLOB         => StringType
       case java.sql.Types.NULL          => null
       case java.sql.Types.NUMERIC       => DecimalType.Unlimited
+      case java.sql.Types.NVARCHAR		  => StringType
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRDD.scala]: do u mind aligning the arrows?

Comment (szheng79) [inline sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRDD.scala]: Fixed, I thought I made aligning, but I guess it caused by mixing space and tab, now I all use tab there.

**Comment by rxin** [comment]
Jenkins, test this please.

**Comment by rxin** [comment]
Thanks. I've merged this.

**Comment by robbyki** [comment]
How can I create a schema outside of spark containing columns with varchar and nvarchar and then save a dataframe with truncate = true and avoid an invalid datatype error for TEXT in Netezza by registering a new dialect? My current dialect has StringType mapped to either varchar or nvarchar but I can't have both and I'm failing to understand how to customize and persists table schemas without my dialects over-writing everything. Using spark 2.1.

**Pull request closed without merging**